### ISSUE TITLE
No error reported when checking `isinstance(x, SomeProtocol)` when `SomeProtocol` is not `@runtime_checkable`

### ISSUE THREAD

**Issue by Tomaz-Vieira** [microsoft/pyright#2151]
**To Reproduce**
```py
# pyright: strict

from typing import Any
from typing_extensions import Protocol

class SomeProtocol(Protocol):
    def f(self) -> int: ...

class MyClass():
    def f(self) -> int:
        return 123


def test(a: Any):
     # TypeError: Instance and class checks can only be used with @runtime protocols
    if isinstance(a, SomeProtocol):
        print(f"a is instance!!")

test(MyClass())
```

**Expected behavior**
Expected an error or warning when checking for `isinstance` on a non `@runtime_checkable` `Protocol`, which is also the mypy behavior

**VS Code extension or command-line**
cli version 1.1.159
pyright extension version 1.1.157

**Comment by erictraut** [comment]
Thanks for the bug report. Pyright does have a check for this, but it was skipped (incorrectly) when the first argument to `isinstance` is of type `Any` or a class that derives from an `Any` or `Unknown` class. I've fixed that problem, so the `runtime_checkable` check is still executed in these cases.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.160, which I just published. It will also be included in the next release of pylance.
